Source/Core/ClippingPlaneCollection.js

```diff
@@ -93,56 +115,146 @@ define([
          * @type {Boolean}
          * @default true
          */
-        combineClippingRegions : {
+        unionClippingRegions : {
             get : function() {
-                return this._combineClippingRegions;
+                return this._unionClippingRegions;
             },
             set : function(value) {
-                if (this._combineClippingRegions !== value) {
-                    this._combineClippingRegions = value;
-                    this._testIntersection = getTestIntersectionFunction(value);
+                if (this._unionClippingRegions !== value) {
+                    this._unionClippingRegions = value;
+                    this._testIntersection = value ? unionIntersectFunction : defaultIntersectFunction;
                 }
             }
         }
     });
 
-    function getTestIntersectionFunction(combineClippingRegions) {
-        if (combineClippingRegions) {
-            return function(value) {
-                return (value === Intersect.INSIDE);
-            };
+    /**
+     *
+     * @param plane
+     *
+     * @exception {DeveloperError} The plane added exceeds the maximum number of supported clipping planes.
+     *
+     * @see ClippingPlaneCollection#remove
+     * @see ClippingPlaneCollection#removeAll
+     */
+    ClippingPlaneCollection.prototype.add = function(plane) {
+        this._planes.push(plane);
+
+        //>>includeStart('debug', pragmas.debug);
```

Comment:
Check for the exception before calling `push`; it leaves the object in a better state.